tsl/src/compression/create.c

```diff
@@ -1244,18 +1034,22 @@ tsl_process_compress_table_add_column(Hypertable *ht, ColumnDef *orig_def)
 		return;
 	}
 
-	Hypertable *compress_ht = ts_hypertable_get_by_id(ht->fd.compressed_hypertable_id);
-	/* don't add column if it already exists */
-	if (get_attnum(compress_ht->main_table_relid, orig_def->colname) != InvalidAttrNumber)
-	{
-		return;
-	}
-
+	List *chunks = ts_chunk_get_by_hypertable_id(ht->fd.compressed_hypertable_id);
+	ListCell *lc;
 	Oid coloid = LookupTypeNameOid(NULL, orig_def->typeName, false);
-	ColumnDef *coldef = build_columndef_singlecolumn(orig_def->colname, coloid);
-	CompressionSettings *settings = ts_compression_settings_get(ht->main_table_relid);
 
-	add_column_to_compression_table(compress_ht, settings, coldef);
+	foreach (lc, chunks)
+	{
+		Chunk *chunk = lfirst(lc);
+		/* don't add column if it already exists */
+		if (get_attnum(chunk->table_id, orig_def->colname) != InvalidAttrNumber)
+		{
+			return;
+		}
```

Comment:
This looks to me as if it should be an error rather than silently ignoring. 

Especially since we no longer have inheritance, it seems to me that there is no implicit enforcement (via inheritance) that a column either exists on all chunks or none of them. I theory, each chunk could now have its own schema with diverging columns. So, if we detect this case it seems to me it should be an error.